### DISCUSSION TITLE
How to run the examples using the jar?

### DISCUSSION THREAD

**Discussion by osa1** [flix/flix#9168] [Q&A] [answered]
I built the .jar file using `gradlew jar` but I cannot figure out how to run the examples in the repo using the jar.

Relevant documentation page (https://doc.flix.dev/getting-started.html#using-flix-from-the-command-line) shows the `run` command to build a project, but the examples directory doesn't seem to be a project? Some of the subdirectories have `flix.toml` files, but most of the examples don't.

`flix.jar` lists the `build` command, but it also requires a `flix.toml` files as far as I understand.

The `repl` command help says "starts a repl for the current project, or provided Flix source files", but in reality it doesn't seem to accept file arguments:

```
$ java -jar ./build/libs/flix.jar repl examples/effects/InputOutput.flix
The 'repl' command cannot be used with a list of files.
```

`:help` in the REPL doesn't list a command to load a source file.

The `examples/` directory also doesn't have a README.

How do I run the examples?

**Comment by magnus-madsen** [comment]
Hi @osa1 and welcome.

Do you want to use VSCode or just the terminal?

**Reply by osa1** [reply]
I want to run it from the shell as it gives me the flexibility to script, and use other editor/IDEs.

**Reply by magnus-madsen** [reply]
OK-- totally understandable. But do give VSCode a try. We have put a lot of effort into it :)

**Comment by magnus-madsen** [comment, marked as answer]
Indeed the examples are not a project. What you do is to create an empty folder, run `java -jar flix.jar init` in the folder, and then copy the files you care about to the `src` directory. That would be one way.

**Reply by osa1** [reply]
Thanks, this works.
